packages/flutter/lib/fix_data/fix_material/fix_widget_state.yaml

```diff
@@ -0,0 +1,196 @@
+# Copyright 2014 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# For details regarding the *Flutter Fix* feature, see
+# https://flutter.dev/docs/development/tools/flutter-fix
+
+# Please add new fixes to the top of the file, separated by one blank line
+# from other fixes. In a comment, include a link to the PR where the change
+# requiring the fix was made.
+
+# Every fix must be tested. See the flutter/packages/flutter/test_fixes/README.md
+# file for instructions on testing these data driven fixes.
+
+# For documentation about this file format, see
+# https://dart.dev/go/data-driven-fixes.
+
+# * Fixes in this file are for the MaterialState enum and classes from the Material library. *
+#   For fixes to
+#     * AppBar: fix_app_bar.yaml
+#     * AppBarTheme: fix_app_bar_theme.yaml
+#     * ColorScheme: fix_color_scheme.yaml
+#     * Material (general): fix_material.yaml
+#     * SliverAppBar: fix_sliver_app_bar.yaml
+#     * TextTheme: fix_text_theme.yaml
+#     * ThemeDate: fix_theme_data.yaml
+version: 1
+transforms:
+  # Changes made in https://github.com/flutter/flutter/pull/142151
+  - title: "Replace with 'WidgetState'"
+    date: 2024-02-01
+    element:
+      uris: [ 'material.dart' ]
+      enum: 'MaterialState'
+    changes:
+      - kind: 'rename'
+        newName: 'WidgetState'
+
+  # Changes made in https://github.com/flutter/flutter/pull/142151
+  - title: "Replace with 'WidgetPropertyResolver'"
+    date: 2024-02-01
+    element:
+      uris: [ 'material.dart' ]
+      function: 'MaterialPropertyResolver'
+    changes:
+      - kind: 'replacedBy'
+        newElement:
+          uris: [ 'widgets.dart' ]
+          function: 'WidgetPropertyResolver'
```

Comment:
CC @goderbauer @Piinks The other fixes in this file as is seem to be working, but I'm having trouble with this one. I can't seem to get the included test to pass. I'm assuming the fact that they have a return type attached `<T>` included makes it complicated for the dart fix to pick up. The definition for WidgetPropertyResolver is:
`typedef WidgetPropertyResolver<T> = T Function(Set<WidgetState> states);`

And a line the test is attempting to check is:
`final MaterialPropertyResolver<MouseCursor?> resolveCallback;`

The test is not correctly turning that line into:
`final WidgetPropertyResolver<MouseCursor?> resolveCallback;`

It doesn't look like we've made a flutter fix for something like this before, so have we tried to do something similar and found it isn't possible, or am I missing something?

Reply:
I was able to get this working.